pkg/sources/chunker_test.go

```diff
@@ -121,6 +123,21 @@ func TestNewChunkedReader(t *testing.T) {
 	}
 }
 
+type panicReader struct{}
+
+var _ io.Reader = (*panicReader)(nil)
+
+func (_ panicReader) Read([]byte) (int, error) {
+	panic("panic for testing")
+}
+
+func TestChunkReader_UnderlyingReaderPanics_DoesNotPanic(t *testing.T) {
```

Comment:
should the test verify that the error is returned through the channel?

Reply:
good question! my vote is "no," because imo the specific correct behavior is not obvious, beyond "shouldn't panic" - which is what the test checks right now.